### PR TITLE
Fix `selector-type-no-unknown` false positives for experimental and deprecated HTML tags

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

check the "concrete example" in #7510

> Is there anything in the PR that needs further explanation?

compare https://github.com/sindresorhus/html-tags/blob/main/html-tags.json and https://github.com/sindresorhus/html-tags/commit/d3b931ee8dc7c5bbbe20761ebbe4dad257cf39bc

- `selectlist`: https://github.com/openui/open-ui/issues/773#issuecomment-1664421419
- `listbox`: https://issues.chromium.org/issues/40146374#comment98
- `portal`: https://caniuse.com/mdn-html_elements_portal
- for the rest check https://developer.mozilla.org/en-US/docs/Web/HTML/Element/

I chose not to include the previous names of the tag names that I am introducing; we need to settle #7510 first.
e.g. not `selectmenu`

### CODE REVIEW

File: lib/reference/selectors.cjs [hunk 1]
```diff
@@ -10,21 +10,26 @@ const deprecatedHtmlTypeSelectors = new Set([
...
 	'big',
+	'bgsound',
 	'blink',
 	'center',
 	'content',
 	'dir',
 	'font',
 	'frame',
 	'frameset',
-	'hgroup',
```
Comment (Mouvedia) [inline lib/reference/selectors.cjs:20]: already part of html-tags package
used to be obsolete

File: lib/reference/selectors.cjs [hunk 2]
```diff
@@ -10,21 +10,26 @@ const deprecatedHtmlTypeSelectors = new Set([
...
+	'image',
```
Comment (Mouvedia) [inline lib/reference/selectors.cjs:21]: was silently converted

File: lib/reference/selectors.mjs [hunk 3]
```diff
@@ -7,21 +7,26 @@ const deprecatedHtmlTypeSelectors = new Set([
...
 	'big',
+	'bgsound',
 	'blink',
 	'center',
 	'content',
 	'dir',
 	'font',
 	'frame',
 	'frameset',
-	'hgroup',
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:17]: already part of html-tags package
used to be obsolete

File: lib/reference/selectors.mjs [hunk 4]
```diff
@@ -7,21 +7,26 @@ const deprecatedHtmlTypeSelectors = new Set([
...
+	'image',
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:18]: was silently converted to ``

File: lib/utils/isCustomElement.cjs [hunk 5]
```diff
@@ -31,10 +30,6 @@ function isCustomElement(selector) {
 		return false;
 	}
 
-	if (selectors.htmlTypeSelectors.has(selectorLowerCase)) {
-		return false;
-	}
-
```
Comment (Mouvedia) [inline lib/utils/isCustomElement.cjs:37]: check l20
i.e. HTML tags don't include `-`

File: lib/utils/isCustomElement.mjs [hunk 6]
```diff
@@ -28,10 +27,6 @@ export default function isCustomElement(selector) {
 		return false;
 	}
 
-	if (htmlTypeSelectors.has(selectorLowerCase)) {
-		return false;
-	}
```
Comment (Mouvedia) [inline lib/utils/isCustomElement.mjs:33]: check l20 it's unnecessary
i.e. HTML tag names don't include `-`

File: .changeset/flat-bananas-hug.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
```
Comment (ybiquitous) [inline .changeset/flat-bananas-hug.md:2]: [ask] Is this PR a patch update?

```suggestion
"stylelint": patch
```

**Review by ybiquitous — COMMENTED** [review]
@Mouvedia Thanks for the pull request. Overall LGTM, but I have one question.

File: .changeset/flat-bananas-hug.md (hunk 7 above)
Comment (Mouvedia) [reply .changeset/flat-bananas-hug.md:2]: stylelint will report more errors so it's a minor
see https://stylelint.io/about/semantic-versioning/#minor-release

Comment (ybiquitous) [reply .changeset/flat-bananas-hug.md:2]: So, "false negatives" are correct?

Comment (ybiquitous) [reply .changeset/flat-bananas-hug.md:2]: This PR seems to reduce false errors, for example: 🤔 

```css
fencedframe, listbox, portal, selectlist {}
```

image

Comment (Mouvedia) [reply .changeset/flat-bananas-hug.md:2]: I don't know.
I will put `patch` to be consistent with previous false-positive PRs.

File: .changeset/flat-bananas-hug.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector-type-no-unknown` false positives
```
Comment (ybiquitous) [inline .changeset/flat-bananas-hug.md:5]: [suggest] Just to be sure, does clarifying targets of this change make sense? E.g.

```suggestion
Fixed: `selector-type-no-unknown` false positives for deprecated HTML tags
```

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**